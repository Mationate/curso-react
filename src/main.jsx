import React from 'react';
import './styles.css';
import ReactDOM, { createRoot } from 'react-dom/client';
import { FirstApp } from './FirstApp';
import { HelloWorldApp } from './HelloWorldApp';
import {CounterApp} from './CounterApp';

ReactDOM.createRoot(document.getElementById('root') ).render(
    <React.StrictMode>
        <CounterApp value={0}  />
    </React.StrictMode>
);