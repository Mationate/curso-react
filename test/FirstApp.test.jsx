import { logDOM, render } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp";

describe('pruebas en FirstApp', () => { 
    // test('debe hacer match con el snapshot', () => { 
    //     const title ='Hola, soy mati';
    //     const subtitule= 123;
    //     const {container} =render(<FirstApp title={title} subtitle={subtitule} />)
        
    //     expect(container).toMatchSnapshot();
    
    // })

    test('Debe mostrar el titulo en el h1', () => { 
        const title ='Hola, soy mati';
        const {container, getByText, getByTestId} = render(<FirstApp title={title} />);
        expect(getByText(title)).toBeTruthy();

        expect(getByTestId('test-title').innerHTML).toContain(title);
        // const h1 = container.querySelector('h1')
        // expect(h1.innerHTML).toBe(title); NO PASARIA SI HAY ESPACIOS
        //* SI PASA
        //expect(h1.innerHTML).toContain(title);
     })


     test('debe mostrar el subtitulo enviado por props', () => { 
        const title ='Hola, soy mati';
        const subtitle ='Hola, soy subtitulo';
        const {getAllByText} = render(
            <FirstApp 
                title={title}
                subtitle={subtitle}
             />
        );

             expect(getAllByText(subtitle).length).toBe(2);
      })

 })