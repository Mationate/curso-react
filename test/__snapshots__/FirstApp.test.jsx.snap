// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en FirstApp debe hacer match con el snapshot 1`] = `
<div>
  <div
    data-testid="test-title"
  >
    Hola, soy mati
  </div>
  <p>
    123
  </p>
  <p>
    Mati
  </p>
</div>
`;
